### PR TITLE
muety/update docs

### PR DESCRIPTION
**Pull request by muety**
* Added config example for Caddy reverse proxy
* Updated Java snippet to use modern [HttpClient](https://openjdk.java.net/groups/net/httpclient/intro.html) instead of outdated `HttpUrlConnection`
* Simplied JS snippet a bit

### CODE REVIEW

File: docs/more-pushmsg.md [hunk 1]
```diff
@@ -154,28 +148,34 @@ public class GotifyClient {
...
 
     private final String gotifyUrl;
+    private final HttpClient httpClient;
+    private final ObjectMapper objectMapper;
 
-    public GotifyClient(String gotifyUrl) {
-        this.gotifyUrl = gotifyUrl;
+    public GotifyClient(String baseUrl, String token) {
+        this.gotifyUrl = String.format("%s/message?token=%s", baseUrl, token);
+        this.httpClient = HttpClient.newHttpClient();
```
Comment (jmattheis) [inline docs/more-pushmsg.md:156]: Nice, didn't know java introduced a new http client in java 11.

**Review by jmattheis — APPROVED** [review]
Thanks, there are some formatting errors, I'll fix them before the merge (running `yarn format`).

File: docs/more-pushmsg.md (hunk 1 above)
Comment (muety) [reply docs/more-pushmsg.md:156]: Yes, it has a cleaner API, HTTP/2 support and a few other features.

There's a lot pretty cool things happening to Java recently. I can only recommend taking a look at all the [JEPs](http://openjdk.java.net/projects/jdk) from version 11 to 16.

**Comment by muety** [comment]
> there are some formatting errors, I'll fix them before the merge

Thanks! I've missed that (shouldn't there be an e-mail notification in case of failing build :thinking:).

**Comment by jmattheis** [comment]
I think you get these notifications from failing github actions, this repo still uses travis-ci.

**Pull request merged**